examples/billing/add_billing_setup.rb

```diff
@@ -0,0 +1,175 @@
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This example creates a billing setup for a customer. A billing setup is a link
+# between a payments account and a customer. The new billing setup can either
+# reuse an existing payments account, or create a new payments account with a
+# given payments profile.
+# Billing setups are applicable for clients on monthly invoicing only. See here
+# for details about applying for monthly invoicing:
+# https://support.google.com/google-ads/answer/2375377.
+# In the case of consolidated billing, a payments account is linked to the
+# manager account and is linked to a customer account via a billing setup.
+
+require 'optparse'
+require 'google/ads/google_ads'
+
+def add_billing_setup(customer_id, payments_account_id, payments_profile_id)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  billing_setup = create_billing_setup(
+    client, customer_id, payments_account_id, payments_profile_id)
+
+  set_billing_setup_date_times(client, customer_id, billing_setup)
+
+  # Issues a mutate request to add the billing setup.
+  response = client.service.billing_setup.mutate_billing_setup(
+    customer_id: customer_id,
+    operation: billing_setup
+  )
+
+  puts "Added new billing setup with resource name #{response.result.resource_name}"
+end
+
+def create_billing_setup(
+  client,
+  customer_id,
+  payments_account_id,
+  payments_profile_id)
+  billing_setup = client.operation.create_resource.billing_setup do |b|
+    # Sets the appropriate payments account field.
+    if payments_account_id
+      # If a payments account ID has been provided, set the resource name.
+      # You can list available payments accounts via the PaymentsAccountService's
+      # ListPaymentsAccounts method.
+      b.payments_account = client.path.payments_account(customer_id, payments_account_id)
+    elsif payments_profile_id
+      # Otherwise, create a new payments account by setting the payments account info.
+      # See https://support.google.com/google-ads/answer/7268503 for more information about
+      # payments profiles.
+      b.payments_account_info = client.resource.payments_account_info do |pai|
+        pai.payments_account_name = "Payment Account #{(Time.new.to_f * 1000).to_i}"
+        pai.payments_profile_id = payments_profile_id
+      end
+    else
+      raise "No payments account ID or payments profile ID is provided."
+    end
+  end
+  billing_setup
```

Comment:
You can omit this and just have the client.operation.create_resource block. The return value of the last line in any function in Ruby is automatically returned by the function.

Reply:
Done